pkg/bpf/map_linux.go

```diff
@@ -1054,7 +1054,9 @@ func (m *Map) GetModel() *models.BPFMap {
 func (m *Map) resolveErrors(ctx context.Context) error {
 	started := time.Now()
 
-	m.lock.Lock()
+	if err := m.openAndLock(); err != nil {
```

Comment:
TIL open / close don't need to be balanced; maps can be opened any number of times but should be closed once.

LGTM.